client/src/get-ticket/components/ticket.tsx

```diff
@@ -1,14 +1,29 @@
-import { useState, useRef } from 'react'
+import { useEffect } from 'react'
 import myIcon from '../../assets/Down_Arrow.svg';
 import '../get-ticket.css';
+import API from '../../API/API.tsx';
 
 
-function Ticket() {
+function Ticket(props: any) {
 
-    const [ticketNo, setTicketNo] = useState(110);
-    const [waitingTime, setWaitingTime] = useState(0);
-    const [serviceType, setServiceType] = useState("Service1");
-    const [show, setShow] = useState(false);
+    // get all services only the fistr time you enter the page
+    useEffect(() => {
+        API.getAllServices()
+            .then(services => {
+                props.setServices(services);
+            })
+            .catch(error => {
+                console.error("Error fetching services:", error);
+            });
+    }, []);
+
+    const AllServices = props.services;
+
+    // get serviceName from serviceID
+    const myService = AllServices.find( (service: any) => (
```

Comment:
This shouldn't be done in this way. You need to have the serviceName already when fetching the ticket from the API. This is a bug in the API and should be fixed.

Reply:
As I talked to the BE developer of the team, For simplicity, we can keep it in this way.
Maybe later we change it.